src/index.js

```diff
@@ -64,11 +71,15 @@ class MediaQuery extends React.Component {
     this.removeMql()
 
     this._mql = matchMedia(this.query, values, forceStatic)
+  }
+
+  updateQuery(props) {
+    this.buildMatcher(props)
     this._mql.addListener(this.updateMatches)
     this.updateMatches()
   }
 
-  componentWillUpdate(_, nextState) {
+  UNSAFE_componentWillUpdate(_, nextState) {
```

Comment:
Maybe we should just lose this feature, I don't even remember why we added it in the first place.

Reply:
Maybe we could move the api call to getSnapshotBeforeUpdate? I've never had a use for this feature either. Since it will be removed in react 17, we'll have to break it eventually.

Reply:
@whatknight Found these:

- https://github.com/contra/react-responsive/pull/92
- https://github.com/contra/react-responsive/pull/97

Feels like an anti-pattern with new react stuff, we can remove it.